### ISSUE TITLE
Feature: (custom content) `SMART_ACTION` types for sending worldstates to the client

### ISSUE THREAD

**Issue by ihm-tswow** [azerothcore/azerothcore-wotlk#7837]
### Describe your feature request or suggestion in detail

tldr: we're adding two new `SMART_ACTION_`s for custom content to tswow, based on TrinityCore. If you want to I can PR the same actions here and we can agree on using the same ID to futureproof it. If you don't want that kind of stuff in your own sai, you can safely close this issue because then we wouldn't clash in the first place.

**The problem**
Currently, there is no good way to send worldstates for use in gossip texts without core scripting. This is a bit unfortunate for things like war effort-style custom game events, where you'd probably want to update that along with sending the gossip menu: 
![image](https://user-images.githubusercontent.com/76849026/132997183-623e11f3-daf2-4767-8436-3f43a71c11b6.png)

### Describe a possible solution to your feature or suggestion in detail

The solution we've made are two new `SMART_ACTION` types: 

`SMART_ACTION_SEND_WORLDSTATE` - Calls `Player::SendUpdateWorldState` either with a value stored in the server (`worldstates` table in tc) or a hardcoded value. param1 decides the worldstate id to send, param2=0 if it's read from the worldserver table, or =1 if it's an immediate value, param3 is the immediate value to send if param2==0, otherwise ignored. 
Can be used for sending arbitrary server worldstates to the client when opening a gossip menu. param2 can later be extended to send other types of values if it turns out useful.

`SMART_ACTION_SEND_EVENT_STATE` - Calls `GameEventMgr::SendWorldStateUpdate`, which sends all the condition worldstates to a target player. param1 = the `game_event` to send worldstates for. Can be used for war effort-style gossips as shown above

### Additional context

We're going to be adding this to tswow anyways so thought we'd just check if you also want the patch, would just be annoying if we both ended up implementing it in the end with different ids. Just let me know if you want it and I can make a pr. If we ever add the other custom ones you have, we'll use the same as you do.

**Comment by FrancescoBorzi** [comment]
@ihm-tswow new smart actions/events/targets are always welcome. It is also a good idea to align the same ID with TC when possible.

Also we should also update [Keira3](https://github.com/azerothcore/Keira3), a tool written in your favourite language :)

**Comment by ihm-tswow** [comment]
Some good additional context: https://github.com/TrinityCore/TrinityCore/issues/26907#issuecomment-917940277
